README.md

```diff
@@ -214,6 +214,75 @@ Custom translations of properties can be added by extending the
 (def datasource-options {:jdbc-url "jdbc:sqlite:db/database.db"})
 ```
 
+### ClickHouse example
+`hikari-cp` can be used with [official ClickHouse driver](https://github.com/ClickHouse/clickhouse-java) in following two ways:
+1. Using `:datasource` property:
+- Official ClickHouse JDBC driver's `ClickHouseDataSource` does not have a zero arity constructor.
+- This mandates creating `ClickHouseDataSource` separately and setting `:datasource` property in `datasource-options`
+```clojure
+(ns hikari-cp.example
+  (:require
+    [hikari-cp.core :refer :all]
+    [next.jdbc :as jdbc])
+  (:import
+    [java.util Properties]
+    [com.clickhouse.jdbc ClickHouseDataSource]))
+
+(defonce clickhouse-datasource
+  (let [jdbc-url "jdbc:ch://localhost:8123/test_db"
+        properties (doto
+                     (Properties.)
+                     (.setProperty "user" "username")
+                     (.setProperty "password" "password")
+                     (.setProperty "client_name" "test_client")
+                     (.setProperty "custom_http_params" "max_query_size=1000000"))]
+    (ClickHouseDataSource. jdbc-url properties)))
+
+(def datasource-options
+  {:datasource clickhouse-datasource
+   :connection-timeout 5000
+   :maximum-pool-size 20
+   :max-lifetime 300000
+   :pool-name "clickhouse-conn-pool"})
+
+(defonce datasource 
+  (delay (make-datasource datasource-options)))
+
+(defn -main
+  [& args]
+  (with-open [conn (jdbc/get-connection {:datasource @datasource})]
+    (let [rows (jdbc/execute! conn ["SELECT 0"])]
+      (println rows)))
+  (close-datasource @datasource))
+```
+2. Using `:jdbc-url` and `:driver-class-name` properties:
+```clojure
+(ns hikari-cp.example
+  (:require
+    [hikari-cp.core :refer :all]
+    [next.jdbc :as jdbc]))
+
+(def datasource-options-2
+  {:jdbc-url "jdbc:ch://localhost:8123/test_db"
+   :username "username"
+   :password "password"
+   :driver-class-name "com.clickhouse.jdbc.ClickHouseDriver"
```

Comment:
@fr33m0nk do you think it's worth mapping an `:adapter` from `com.clickhouse.jdbc.ClickHouseDriver` to `:clickhouse`?

See: https://github.com/tomekw/hikari-cp#adapters-and-corresponding-datasource-class-names

Reply:
`com.clickhouse.jdbc.ClickHouseDriver` isn't the DataSource class.
`com.clickhouse.jdbc.ClickHouseDataSource` is the DataSource class, however `com.clickhouse.jdbc.ClickHouseDataSource` doesn't come with default (zero arity) constructor resulting in below exception:
```clojure
Execution error (NoSuchMethodException) at java.lang.Class/getConstructor0 (Class.java:3685).
com.clickhouse.jdbc.ClickHouseDataSource.<init>()
```
If mapping an` :adapter` like `:clickhouse` -> `com.clickhouse.jdbc.ClickHouseDataSource` is still needed, following additional logic would need to be factored in `hikari-cp.core` at around [this place](https://github.com/tomekw/hikari-cp/blob/master/src/hikari_cp/core.clj#L237)

```clojure
(cond
      (and adapter (= adapter :clickhouse))
      (->> (UtilityElf/createInstance datasource-class-name DataSource (to-array [jdbc-url]))
           (.setDataSource config))
      
      adapter (.setDataSourceClassName config datasource-class-name)
      
      :else (.setJdbcUrl config jdbc-url))
```

TBH, this would be customising `hikari-cp` to fit in ClickHouse. I have seen several threads in `Hikari-CP` repo where, the author suggests to use `setDataSource`  instead of `setDataSourceClassName` when DataSource class does not provide default constructor or has non-standard getter and setter. [An example](https://github.com/brettwooldridge/HikariCP/issues/1728)